app/views/manage/mturk_auto/evaluate_batch.html.erb

```diff
@@ -94,39 +94,45 @@
 </div>
 
 <h3>Qualification list</h3>
-<p><i>Last updated: <%= @mturk_batch_job.mturk_worker_qualification_list.updated_at.strftime('%e %b %Y') %></i></p>
-<% mturk_workers = @mturk_batch_job.mturk_worker_qualification_list.mturk_workers %>
-<div class="row justify-content mb-4">
-  <div class="col-6">
-    <table class="table">
-      <thead>
-        <tr>
-          <th>Worker Id</th>
-          <th>Num tweets</th>
-        </tr>
-      </thead>
-      <tbody>
-        <% mturk_workers.each do |mturk_worker| %>
+<% if @mturk_batch_job.mturk_worker_qualification_list.nil? %>
+  <p>No qualification list was indicated for this batch. A new qualification list for the future batches will be formed by evaluating the results.</p>
+<% else %>
+  <p><i>Last updated: <%= @mturk_batch_job.mturk_worker_qualification_list.updated_at.strftime('%e %b %Y') %></i></p>
+  <% mturk_workers = @mturk_batch_job.mturk_worker_qualification_list.mturk_workers %>
+  <div class="row justify-content mb-4">
+    <div class="col-6">
+      <table class="table">
+        <thead>
           <tr>
-            <td><%= mturk_worker.worker_id %></td>
-            <td><%= mturk_worker.mturk_tweets.count %></td>
+            <th>Worker Id</th>
+            <th>Num tweets</th>
           </tr>
-        <% end %>
-      </tbody>
-    </table>
+        </thead>
+        <tbody>
+          <% mturk_workers.each do |mturk_worker| %>
+            <tr>
+              <td><%= mturk_worker.worker_id %></td>
+              <td><%= mturk_worker.mturk_tweets.count %></td>
+            </tr>
+          <% end %>
+        </tbody>
+      </table>
+    </div>
   </div>
-</div>
+<% end %><br/>
 
 <% unless @mturk_auto_batch.evaluated? %>
-  <h3 class="mb-3">Update qualification list</h4>
-  <%= simple_form_for(@mturk_batch_job.mturk_worker_qualification_list) do |f| %>
+  <% new_ql = @mturk_batch_job.mturk_worker_qualification_list.nil? %>
+  <h3><%= new_ql ? 'Create a' : 'Update' %> qualification list</h3>
+  <% mturk_worker_qualification_list = @mturk_batch_job.mturk_worker_qualification_list || MturkWorkerQualificationList.new %>
+  <%= simple_form_for(mturk_worker_qualification_list) do |f| %>
     <%= f.error_notification %>
     <div class="form-group">
       <%= f.error :base %>
     </div>
 
     <div class="form-group">
-      <%= f.input :name, hint: 'Short name of the qualification list. Has to be unique. Is public to workers.', readonly: true %>
+      <%= f.input :name, hint: 'Short name of the qualification list. Has to be unique. Is public to workers.', readonly: !mturk_worker_qualification_list.name.nil? %>
```

Comment:
Is `!mturk_worker_qualification_list.name.nil?` the best filter we have? I see that there is presence validation in the model, but the DB does not have a NOT NULL constraint (DB also does not have a unique index, which makes the uniqueness validation inefficient) so this could fail unexpectedly. Fixing DB integrity could be done in separate issues, but here why not just check `f.object.persisted?`?

Reply:
Yeah, nice suggestion, thanks! `f.object.persisted` is what I really need here. Fixing DB integrity is one of the issues that I never really approached, but it would be nice to go through it with you before I leave, since I'd like to hear your feedback on it just to learn a few things about it :)